torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -667,8 +667,12 @@ at::Tensor PythonArgs::tensor_slow(int i) {
     throw TypeError("expected Tensor as argument %d, but got %s", i,
         Py_TYPE(obj)->tp_name);
   }
-  auto tensor = scalar_to_tensor(scalar);
-  tensor.unsafeGetTensorImpl()->set_wrapped_number(true);
+  at::Tensor tensor;
+  {
+    at::AutoNonVariableTypeMode guard_;
+    tensor = scalar_to_tensor(scalar);
+    tensor.unsafeGetTensorImpl()->set_wrapped_number(true);
+  }
   return autograd::make_variable(tensor);
```

Comment:
The `autograd::make_variable()` is kept here but removed from the change to `internal_new_from_data()` in `tensor_new.cpp`.
Is there a reason? Should we keep using `make_variable()` going forward?

Reply:
I think every call that looks like `make_variable(x)` can be soundly replaced with `x`, but I haven't carefully verified this. In `internal_new_from_data` it looks like I accidentally deleted the `make_variable` (but it looks like it didn't make a difference)

Reply:
I didn't look into `make_variable` in great details but it seems to be doing non-trivial stuff. No?

Reply:
Most of what it is doing is making a shallow copy of the tensor and setting some properties (like `requires_grad`) appropriately on the shallow tensor. Some of the complication is that when you only have one reference to the tensor, you just do that modification in-place.

In the new world, I think most occurrences of `make_variable(x)` with no arguments are a no-op, because `x` is supposed to not require grad already. Occurrences of `make_variable(x, /*requires_grad*/ true)` probably could have been replaced by just by initializing the variable correctly in the first place. It's hard to say and will need to be done on a case-by-case basis.

Reply:
Sounds good.